miden/src/tools/mod.rs

```diff
@@ -0,0 +1,61 @@
+use assembly::Assembler;
+use vm_core::{Operation, ProgramInputs};
+
+#[derive(Debug, PartialEq)]
+pub struct ProgramInfo {
+    total_vm_cycles: usize,
+    total_noops: usize,
+}
+
+impl ProgramInfo {
+    ///Creates a new ProgramInfo object
+    pub fn new(total_vm_cycles: usize, total_noops: usize) -> ProgramInfo {
+        Self {
+            total_vm_cycles,
+            total_noops,
+        }
+    }
+
+    ///Get total vm cycles to execute a program
+    pub fn total_vm_cycles(&self) -> usize {
+        self.total_vm_cycles
+    }
+
+    ///Get total noops executed as part of a program
+    pub fn total_noops(&self) -> usize {
+        self.total_noops
+    }
+}
+
+/// Returns program analysis for a given script string. Returns ProgramInfo with following fields:
+/// - total_vm_cycles (vm cycles it takes to execute the entire script)
+/// - total_noops (total noops executed as part of a program)
+pub fn analyze(program: &str, inputs: ProgramInputs) -> ProgramInfo {
+    let assembler = Assembler::new();
+    let script = assembler.compile_script(program).unwrap();
+    let vm_state_iterator = processor::execute_iter(&script, &inputs);
+
+    let mut total_vm_cycles = 0;
+    let mut noop_count = 0;
+
+    for vm_state in vm_state_iterator {
+        if vm_state.as_ref().unwrap().op == Some(Operation::Noop) {
```

Comment:
I think you should be able to change this to: `if matches!(op, Operation::Noop) {`.

Reply:
Thanks. Done.